Feature/integrate sdc rabbit

gemmai95 opened the pull request:
## What? and Why?
> What does this pull request change/fix? Why was it necessary?

Integrate with common sdc rabbit library to remove code duplication and margin for error.

## Checklist
  - [x] CHANGELOG.md updated? (if required)

gemmai95 committed:
Integrate sdc rabbit library

gemmai95 committed:
Update changelog

gemmai95 committed:
Add unit tests for exceptions

gemmai95 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [sdx-receipt-rrm to use sdc-rabbit library (3)](https://trello.com/c/Y102Kw0s/297-sdx-receipt-rrm-to-use-sdc-rabbit-library-3)

gemmai95 committed:
Remove unused import

tej99 commented on app/main.py:
```diff
@@ -0,0 +1,45 @@
+
+
+def run():
+    logging.getLogger("pika").setLevel(logging.INFO)
```
Need to remove setting log level for pika logger, causes problems with picking up logs in splunk.

tej99 commented on app/response_processor.py:
```diff
@@ -79,7 +91,7 @@ def _send_receipt(self, decrypted, xml):
             if res.status_code == 400:
                 res_logger.error("Receipt rejected by endpoint")
-                raise BadMessageError("Receipt rejected by endpoint")
+                raise ClientError
```
Looks like ClientError is not getting handled

gemmai95 committed:
Remove pika logging level and handle client error

gemmai95 committed:
Remove client error

necrophonic committed:
Null-change - build trigger

Added a blank line to readme to attempt trigger of codecov build without affecting code.
